modules/react/common/lib/utils/elements.ts

```diff
@@ -43,15 +43,33 @@ export const getFirstFocusableElement = (container: HTMLElement): HTMLElement |
   return null;
 };
 
+const isRadioInput = (element: Element): element is HTMLInputElement => {
+  return element.nodeName.toLowerCase() === 'input' && element.getAttribute('type') === 'radio';
+};
+
+const getRadioGroup = (
+  container: HTMLElement,
+  element: HTMLInputElement
+): NodeListOf<HTMLInputElement> => {
+  return container.querySelectorAll(`input[type="radio"][name="${element.getAttribute('name')}"]`);
+};
+
 /**
  * Get the last focusable element in a container.
+ *
+ * Returns an array of elements if the last focusable element is a radio group
  */
-export const getLastFocusableElement = (container: HTMLElement): HTMLElement | null => {
+export const getLastFocusableElement = (container: HTMLElement): Element[] | Element | null => {
   const elements = container.querySelectorAll('*');
 
   for (let i = elements.length - 1; i >= 0; i--) {
     const element = elements.item(i);
     if (element && isFocusable(element) && element.getAttribute('tabindex') !== '-1') {
+      if (isRadioInput(element)) {
+        const radioGroup = getRadioGroup(container, element);
+
+        return radioGroup.length > 0 ? Array.from(radioGroup) : element;
+      }
```

Comment:
Why not instead update `isFocusable` to exclude `input[type=radio]` that isn't checked?

Reply:
I guess that wouldn't work because if non of the radios in the group are `checked`, none of them would be considered focusable.

Reply:
> I guess that wouldn't work because if non of the radios in the group are checked, none of them would be considered focusable.
Yes, it would not in that case.

Also if onChange is being slow (say, it invokes something else before finally updating radio group's value which causes the input to become selected), a certain input stays focused and not yet selected for a few frames. It's a developer to blame in that case, but we can guard this corner case too